### PR TITLE
Pti balance

### PR DESCRIPTION
**Pull request by bent0b0x**
**Work Done**
- show PTI balance in nav bar

image

**NOTE:** the balance is rendered by a connected container that refreshes its own value every 2 seconds (easily configurable). Let me know if you have any objections/alternative suggestions on a better way to approach this.

cc @pedrocasa @felipegaucho

### CODE REVIEW

File: src/scripts/components/widgets/PTIBalance.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+`
+
+const REFRESH_BALANCES_INTERVAL_MS: number = 2000
+
+class PTIBalance extends React.Component<Props, void> {
+  intervalId: ?number
+
+  componentWillMount = (): void => {
+    const { loadBalances } = this.props
+    this.intervalId = setInterval(loadBalances, REFRESH_BALANCES_INTERVAL_MS)
```
Comment (bent0b0x) [inline src/scripts/components/widgets/PTIBalance.js]: The component refreshes its own value every 2 seconds

File: src/scripts/components/widgets/PTIBalance.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  componentWillUnmount = (): void => {
+    if (this.intervalId) {
+      clearInterval(this.intervalId)
```
Comment (bent0b0x) [inline src/scripts/components/widgets/PTIBalance.js]: Cleans itself up when unmounted

File: test/functional-tests/app.spec.js [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
+      browser.waitUntil(() => {
+        return browser.getTitle() === 'Paratii'
+      })
+    })
+  })
+  describe('Nav bar', () => {
+    it("should render the user's PTI balance", () => {
+      browser.url('http://localhost:8080')
+      browser.waitUntil(
+        () => browser.getText('[data-test-id="nav-pti-balance"]') === '0 PTI'
```
Comment (bent0b0x) [inline test/functional-tests/app.spec.js]: @jellegerbrandy suggestions on how to set up a test where the user has a balance other than `0`?

**Comment by felipegaucho** [comment]
Awesome (despite me not knowing exactly what .5b PTI means) :)

**Comment by bent0b0x** [comment]
The person in the screenshot is rich. 39.5 Billion.

**Pull request merged**